### PR TITLE
[httplib, requests] Sanitize urls in span metadata

### PR DESCRIPTION
**Pull request by majorgreys**
Use `urlparse` in Python 2 and Python 3 standard library to only set the path (without parameters and query string) in the `http.url` metadata.

### CODE REVIEW

File: ddtrace/compat.py [hunk 1]
```diff
@@ -22,6 +23,7 @@
 else:
     from queue import Queue
     from urllib.parse import urlencode
+    from urllib.parse import urlparse
```
Comment (brettlangdon) [inline ddtrace/compat.py]: We have this down below already.

```python
from ...compat import parse

parse.urlparse(url)
```

Comment (majorgreys) [reply ddtrace/compat.py]: Right, hadn't noticed it before. Fixed now

File: tests/contrib/httplib/test_httplib.py [hunk 2]
```diff
@@ -242,7 +241,7 @@ def test_httplib_request_get_request_query_string(self):
             {
                 'http.method': 'GET',
                 'http.status_code': '200',
-                'http.url': '{}?key=value&key2=value2'.format(URL_200),
+                'http.url': '{}'.format(URL_200),
```
[the remark this answers is not in the document]
Comment (majorgreys) [reply tests/contrib/httplib/test_httplib.py]: Added a comment to the test assertion. That address your point?

File: ddtrace/contrib/requests/connection.py [hunk 3]
```diff
@@ -55,6 +53,15 @@ def _wrap_request(func, instance, args, kwargs):
     url = kwargs.get('url') or args[1]
     headers = kwargs.get('headers', {})
     parsed_uri = parse.urlparse(url)
+    # sanitize url of query
+    sanitized_url = parse.urlunparse((
```
Comment (brettlangdon) [inline ddtrace/contrib/requests/connection.py]: :+1:

I was originally thinking just `parsed_uri.path` or something, but this is much better.

File: ddtrace/contrib/httplib/patch.py [hunk 4]
```diff
@@ -60,9 +59,14 @@ def _wrap_putrequest(func, instance, args, kwargs):
         method, path = args[:2]
         scheme = 'https' if isinstance(instance, httplib.HTTPSConnection) else 'http'
         port = ':{port}'.format(port=instance.port)
+
+        # strip path of non-path data
+        path = parse.urlparse(path).path
+
         if (scheme == 'http' and instance.port == 80) or (scheme == 'https' and instance.port == 443):
             port = ''
         url = '{scheme}://{host}{port}{path}'.format(scheme=scheme, host=instance.host, port=port, path=path)
```
Comment (brettlangdon) [inline ddtrace/contrib/httplib/patch.py]: Should we use `parse.urlunparse(())` like we did below?

File: tests/contrib/httplib/test_httplib.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (brettlangdon) [reply tests/contrib/httplib/test_httplib.py]: yup, the test assertion just said that it tested that the query string was included, which was changing.

File: tests/contrib/httplib/test_httplib.py [hunk 5]
```diff
@@ -242,7 +241,8 @@ def test_httplib_request_get_request_query_string(self):
             {
                 'http.method': 'GET',
                 'http.status_code': '200',
-                'http.url': '{}?key=value&key2=value2'.format(URL_200),
+                # check url metadata lacks query string
```
Comment (brettlangdon) [inline tests/contrib/httplib/test_httplib.py]: oh, sorry, it is L223 in this file that needs to change:

`we capture the entire url in the span`

which is not longer true.

File: ddtrace/contrib/httplib/patch.py (hunk 4 above)
Comment (brettlangdon) [reply ddtrace/contrib/httplib/patch.py]: this is not a required change, just an observation

**Comment by brettlangdon** [comment]
might need to update from `0.16-dev` first

File: ddtrace/contrib/httplib/patch.py (hunk 4 above)
Comment (majorgreys) [reply ddtrace/contrib/httplib/patch.py]: for now copy-and-pasted code from requests though we should consider down the line this as a good function for metadata sanitization.

**Pull request merged**